cmd/operator-sdk/bundle/output.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+)
+
+const (
+	infoLevel  = "INFO"
+	errorLevel = "ERROR"
+)
+
+// Log is an object which is used to return results in the JSON format
+type Log struct {
+	Type    string `outputFormat:"type"`
+	Message string `outputFormat:"message"`
+	Error   string `outputFormat:"error,omitempty"`
+}
+
+// OutPut represents the final result
+type OutPut struct {
+	Passed bool  `outputFormat:"passed"`
+	Logs   []Log `outputFormat:"logs"`
+}
+
+// newLog return a new *Log based on the info provided
+func newLog(msg string, err error) *Log {
+	log := Log{}
+	log.Type = infoLevel
+	log.Message = msg
+	if err != nil {
+		log.Error = err.Error()
+		log.Type = errorLevel
+	}
+	return &log
+}
+
+// printJSON will print the output in JSON format
+func (o *OutPut) printJSON() {
+	o.Passed = true
+	for _, i := range o.Logs {
+		if o.Passed && i.Type == errorLevel {
+			o.Passed = false
+			break
+		}
+	}
+	prettyJSON, err := json.MarshalIndent(o, "", "    ")
+	if err != nil {
+		log.Fatal("Failed to generate outputFormat", err)
+	}
+	fmt.Printf("\n%s\n", string(prettyJSON))
```

Comment:
We can remove the first one but we need to have the \n at the end. Otherwise, the result will be:
<img width="182" alt="Screenshot 2020-05-15 at 19 13 39" src="https://user-images.githubusercontent.com/7708031/82082598-34267e00-96e0-11ea-8973-ef0d49d5f406.png">

Reply:
That's fine since the output is being read by a machine.

Reply:
Could ouput it more human readable, as you described "pretty" be a problem? I mean, could it cause any problem to justify we deliberated do not? if not, I think we can move forward with.

Reply:
Removed .. to use the logger lib instead of.